### PR TITLE
[SPIRVToLLVM] Convert `spirv.ControlBarrier`

### PR DESCRIPTION
**Pull request by whitneywhtsang**
Note that this is a temporary solution.

### CODE REVIEW

File: mlir/lib/Conversion/SPIRVToLLVM/SPIRVToLLVM.cpp [hunk 1]
```diff
@@ -1492,6 +1492,43 @@ class VectorShufflePattern
...
+
+class ControlBarrierPattern
+    : public SPIRVToLLVMConversion<spirv::ControlBarrierOp> {
+public:
+  using SPIRVToLLVMConversion<spirv::ControlBarrierOp>::SPIRVToLLVMConversion;
+  LogicalResult
+  matchAndRewrite(spirv::ControlBarrierOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    auto i32Ty = IntegerType::get(rewriter.getContext(), 32);
```
Comment (etiotto) [inline mlir/lib/Conversion/SPIRVToLLVM/SPIRVToLLVM.cpp:1508]: auto - > Type

**Review by etiotto — APPROVED** [review]
Minor comment left inline. 

Noting here that this is a temporary solution because we should lower `gpu.barrier` to a GENX dialect operation rather than a SPIRV dialect operation.

File: mlir/lib/Conversion/SPIRVToLLVM/SPIRVToLLVM.cpp (hunk 1 above)
Comment (whitneywhtsang) [reply mlir/lib/Conversion/SPIRVToLLVM/SPIRVToLLVM.cpp:1508]: `IntegerType::get` should return `IntegerType`, I think we can use `auto` in this case.

Comment (pengtu) [reply mlir/lib/Conversion/SPIRVToLLVM/SPIRVToLLVM.cpp:1508]: Agree with Ettore that this is a temporary solution.  We shall start moving Triton MLIR GPU -> MLIR SPIRV to Triton MLIR GPU -> GENX MLIR so we don't do unnecessary work.  We can define a genx.barrier in GENX and use same SPIRV device barrier function to lower it to LLVM.

**Review by pengtu — APPROVED** [review]

File: mlir/lib/Conversion/SPIRVToLLVM/SPIRVToLLVM.cpp (hunk 1 above)
Comment (whitneywhtsang) [reply mlir/lib/Conversion/SPIRVToLLVM/SPIRVToLLVM.cpp:1508]: Agree this is temporary solution, added to PR description.

**Pull request merged**